README.md

```diff
@@ -4,45 +4,145 @@
 
 # QBoost
 
-The D-Wave quantum computer has been widely studied as a discrete optimization
-engine that accepts any problem formulated as quadratic unconstrained binary
-optimization (QUBO). In 2008, Google and D-Wave published a paper, [Training a
-Binary Classifier with the Quantum Adiabatic
-Algorithm](https://arxiv.org/pdf/0811.0416.pdf), which describes how the QBoost
-ensemble method makes binary classification amenable to quantum computing: the
-problem is formulated as a thresholded linear superposition of a set of weak
-classifiers and the D-Wave quantum computer is used to optimize the weights in a
-learning process that strives to minimize the training error and number of weak
-classifiers
-
-This code demonstrates the use of the D-Wave system to solve a binary
-classification problem using the QBoost algorithm.
+In machine learning, boosting methods are used to combine a set of simple,
+"weak" predictors in such a way as to produce a more powerful, "strong"
+predictor.  QBoost is an approach for boosting that involves selecting weak
+classifiers based on minimization of quadratic loss (squared difference between
+the observed and predicted label), so that the resulting optimization problem
+can be formulated as a binary quadratic model and solved on the D-Wave system.
+This approach has been explored previously in the literature, for example
+Refs. [1-4].
 
-## Disclaimer
-
-This demo and its code are intended for demonstrative purposes only and are not
-designed for performance.
+This example illustrates the application of a QBoost formulation towards two
+illustrative data sets.
 
 ## Usage
 
-To run a demonstration using one of two provided sample data sets, run:
+The demonstration includes two sample data sets, which can be run with the
+following commands:
 
 ```bash
-python demo.py  wisc
+python demo.py blobs
 ```
 
 Or:
 
 ```bash
-python demo.py mnist
+python demo.py digits
+```
```

Comment:
Yes, there is a tradeoff here.  The approach that I took creates sub-commands for the data sets.  The advantage is that this groups together command options that are only applicable to a specific sub-command (data set).  The drawback is that `argparser` doesn't show the options for these sub-commands when you run the top-level help (I wish it did).  But that is why I explicitly made a note in the README about how to get the options that are specific to each data set.

The alternative would be to make the data set name a positional argument instead of a sub-command.  The drawback of this is that there is no programmatic checking or grouping of the options with the data set.  We could do this, but then we probably need to expand each help message, e.g.:
```
--num-samples NUM_SAMPLES
         number of samples (default: %(default)s).  Only applicable for "blobs" data
```
And then also do we want to raise an error if the user invokes an invalid argument combination (e.g., with something like `python demo.py digits --num-samples=100`)?  Sub-commands let `argparse` handle this for us.